std/typecons.d

```diff
@@ -1802,6 +1803,13 @@ private template ReverseTupleSpecs(T...)
     );
 }
 
+// Issue 8494, parte uno
+@safe unittest
+{
+    auto a = tuple(3, "foo");
+    a = (a = a);
```

Comment:
This should be written as `assert(__traits(compiles, { a = (a = a); }));` Generally, the unit tests for phobos are checkable predicates.

Also, there should be a test case for aliasing, e.g.:

```
Tuple!(int[]) a, b, c;
a = tuple([0, 1, 2]);
c = b = a;
assert(a.length == b.length && b.length == c.length);
assert(a.ptr == b.ptr && b.ptr == c.ptr);
```

Reply:
> This should be written as assert(__traits(compiles, { a = (a = a); }));

No, your later example is much better. `__traits(compiles, ...)` is an anti-pattern in unittest, because it doesn't check that the result is actually correct. There can also be subtle differences between CT and RT execution, which can also skew the result of this check. There was an effort a couple of months ago to remove many instances of `__traits(compiles, ...)` from phobos.

Reply:
Sorry, I went to reply and accidentally edited your comment instead.
------------------------------------------------------------------------

> No, your later example is much better. `__traits(compiles, ...)` is an anti-pattern in unittest...

That's exactly what was being tested though, whether `a = (a = a)` can compile. If that wasn't the intention then I agree that the test should be changed.

> There was an effort a couple of months ago to remove many instances of __traits(compiles, ...) from phobos.

Depending on what it was replaced with, the cure could be worse than the disease. `is(typeof(...))` is the other construct that was heavily used for testing if something compiles historically, but is far worse than using `__traits(compiles)` because it returns false for a myriad of other cases than just if an expression doesn't compile.